#167729779 Setup code coverage

davidfrank96 opened the pull request:
#### What does this PR do?
This is the setup for coveralls for code coverage 
#### Description of Task to be completed?
- implemented scripts for coveralls.io to use
- implemented a coveralls file
- included a repo token in the coveralls.yml file
#### How should this be manually tested?
N/A
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#167729779
#### Screenshots (if appropriate)
#### Questions:

davidfrank96 committed:
chore(coveralls): implement exactly once delivery
- implemented scripts for coveralls.io to use
- implemented a coveralls file
[Delivers #130635935]

davidfrank96 committed:
chore(coverals): implement exactly once delivery
- included a repo token in the coverals.yml file
[Delivers #167729779]

Stylll commented on .coverals.yml:
```diff
@@ -0,0 +1 @@
+repo_token: bsXgjuI45eiXHPks5pUWcade8RGeiQehz
```
Hey @davidfrank96, we should avoid saving our repo token on Github. Anyone who can view this repo will be able to use this token at free will. You can remove this file from git tracking and add a sample file showing the necessary values to be added in the `.coverals.yml` file. You should also update the Readme so future collaborators can know they need to create a `.coverals.yml` locally

davidfrank96 replied:
I have included the .coveralls.yml file in my 

`.gitignore`

 file and secondly have  updated my `READme.rd` file

Stylll replied:
@davidfrank96 Even though you have added this file in `.gitignore`, git still tracks it so you will have to remove it from git tracking so it doesn't appear in your PR

davidfrank96 replied:
okay.. done that already... still surprised why that is..but working on it now

davidfrank96 replied:
So sorry about this... I fixed this but got a review from my team members to change the spelling of coveralls so that tracked it back to the repo

Stylll commented on package.json:
```diff
@@ -8,7 +8,11 @@
     "dev": "nodemon --exec babel-node src/index.js",
     "lint": "eslint src/ --fix",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
```
@davidfrank96 there should be a valid test script here as that's how you can run test and confirm that the coverage report works fine

davidfrank96 replied:
Ignored this...since you said it's not necessary that Chima will fix it...

davidfrank96 committed:
chore(Coveralls): implement exactly once delivery
- included the coveralls badge in the readme
- Changed repo token
[Delivers #167729779]

davidfrank96 committed:
chore(Coverals): implement exactly once delivery
- Added code climates badges to the readMe file
-Included badges for code maintainability and for test coverage
[Delivers #167729779]

mthamayor reviewed (commented):
Testing slack integrations

mthamayor reviewed (commented):
I'm sorry. Testing again

davidfrank96 committed:
chore(coveralls): implement exactly once delivery
- included the .coveralls.yml file in .gitignore file
- Updated my README.md file to get the coveralls.yml file locally
- Insatlled coveralls and nyc as a dev dependency
[Delivers #167729779]

davidfrank96 committed:
Chore(Coveralls): implement exactly once delivery
- Resolved a code conflicts on my READme.md after merging locally
[Delivers #130635935]

davidfrank96 committed:
Chore(gitcommit)
- corrected my commit message with the right pivotal tracker ID
[Delivers #167729779]

davidfrank96 committed:
chore(gitignore): implement exactly once delivery
- Unstage everything in the index file , so i can add the coveralls.yml file
[Delivers #167729779]

sebaztyn reviewed (approved):
Great work bro. Kindly edit your pull request - the pivotal tracker relevant story should be a link to the chore, you used its story ID.

mthamayor reviewed (commented):
Please make your PR message more descriptive. Something like #167729779 Setup code coverage
And I requested changes on the spelling of `.coveralls.yml`. This is still present here https://github.com/andela/shadowcat-backend/blob/2478072522a8a033696d332b6b7663ce76090ba6/.gitignore#L102
Will there not be a problem with coveralls not being able to pick up its config file because of wrong spelling?

okabamac reviewed (approved):
Looks good, Frank

Stylll commented on README.md:
````diff
@@ -34,3 +34,4 @@ Make company global travel and accommodation easy and convenient for the strong
     npm run start
   ```
 
+- You need to get a .coveralls.yml file locally for code coverage to get started
````
@davidfrank96 I see that you've added a message here. This should have a title and also explain how to go about getting the token to put in the `.coveralls.yml` file

davidfrank96 replied:
Done..

Stylll reviewed (changes requested):
Hey @davidfrank96, I dropped some comments. Kindly check them out.

davidfrank96 committed:
chore(Coverals)
- Corrected coverals.yml to .coveralls.yml
[Finishes #167729779]

davidfrank96 committed:
chore(bugfix)
- included the steps for getting the repo_token for coveralls in the READme.md file
-Untracked the coveralls.yml file from to gitignore file
[finishes #167729779]

Stylll commented on README.md:
```diff
@@ -34,3 +34,8 @@ Make company global travel and accommodation easy and convenient for the strong
+ Coveralls for code coverage
+- Clone the repo
+- create a file .coveralls.yml
+- Untrack the `repo_token` from .gitignore file
```
@davidfrank96  `Untrack the `repo_token` from .gitignore file`. I don't understand what this means. You can replace it with how the user can get the `repo_token` and how to put it in the `.coveralls.yml` file

Stylll reviewed (changes requested):
Tiny change!

davidfrank96 committed:
chore(repotoken)
- Describe the process of a user to the started with coveralls and the repo_token
[finishes #167729779]

davidfrank96 committed:
chore(gitignore)
- inclued coveralls icov.info file in the gitignore
[Finishes #167729779]

davidfrank96 committed:
chore(icov.info)
- removing . from lcov.info from gitignore file
[finishes #167729779]
